scripts/bee.sh

```diff
@@ -25,30 +25,50 @@ USAGE
     exit 1
 }
 
-stop() {
+stop_containers() {
     echo "Stop Bee following containers:"
     docker container stop "$QUEEN_CONTAINER_NAME";
     WORKER_NAMES=$(docker container ls -f name="$WORKER_CONTAINER_NAME*" --format "{{.Names}}")
     for WORKER_NAME in $WORKER_NAMES; do
         docker container stop "$WORKER_NAME"
     done
+}
 
+stop() {
+    stop_containers
     trap - SIGINT
     exit 0;
 }
 
+queen_failure() {
+    docker logs "$QUEEN_CONTAINER_NAME"
+    stop_containers
+    echo "Timeout limit has been reached, exit from the process.."
+    exit 1
+}
+
 fetch_queen_underlay_addr() {
     if [[ -n "$QUEEN_UNDERLAY_ADDRESS" ]] ; then return; fi
 
-    while : ; do
-        QUEEN_UNDERLAY_ADDRESS=$(curl -s localhost:1635/addresses | python -mjson.tool 2>&1 | grep "/ip4/" | awk '!/127.0.0.1/' | sed 's/,$//' | xargs)
+    ELAPSED_TIME=0
+    WAITING_TIME=5
+    # Wait 2 mins for queen start
+    TIMEOUT=$((2*12*WAITING_TIME))
+    while (( TIMEOUT > ELAPSED_TIME )) ; do
+        QUEEN_UNDERLAY_ADDRESS=$(curl -s localhost:1635/addresses | python -mjson.tool | grep "/ip4/" | awk '!/127.0.0.1/' | sed 's/,$//' | xargs)
         if [[ -z "$QUEEN_UNDERLAY_ADDRESS" ]] ; then
             echo "Waiting for the Queen initialization..."
-            sleep 5
+            ELAPSED_TIME=$((ELAPSED_TIME+WAITING_TIME))
+            sleep $WAITING_TIME
         else
+            echo "Queen underlay address: $QUEEN_UNDERLAY_ADDRESS"
             break;
         fi
     done
+    
+    if (( TIMEOUT == ELAPSED_TIME )) ; then
```

Comment:
I think it would be better to use `<=` instead of strict equality.

Reply:
Why? the `TIMEOUT` is a multiple of `WAITING_TIME`

Reply:
Currently it is, but it is not documented that this is a requirement. If in the future someone changes the code it may break. Actually thinking about this I would write 
```typescript
if (( ELAPSED_TIME >= TIMEOUT )) ; then
```
because that would express the logical condition here the best (at least for me).